bazel_integration_test/private/bazel_integration_test.bzl

```diff
@@ -89,13 +90,25 @@ def bazel_integration_test(
                      values.
         additional_env_inherit: Optional. Specify additional `env_inherit`
                                 values that should be passed to the test.
+        bazel_binaries: Optional for WORKSPACE loaded repositories. Required
+            for repositories that enable bzlmod. The value for this parameter
+            is loaded by adding
+            `load("@bazel_binaries//:defs.bzl", "bazel_binaries")` to your
+            build file.
```

Comment:
Is this ever expected to be something else than `@bazel_binaries//:defs.bzl%bazel_binaries`? If not, maybe this could also be loaded right here. The fallback for the WORKSPACE mode could also be achieved by a repository rule that generates `@bazel_binaries` in WORKSPACE mode.

Reply:
It may not always be every Bazel version that was loaded. For instance, if a client defines three Bazel versions, the test may only be relevant for two of the three Bazel binaries. This repo has some trivial examples demonstrating this. In practice, it almost always is the full list of Bazel versions.

Reply:
> For instance, if a client defines three Bazel versions, the test may only be relevant for two of the three Bazel binaries.

But isn't that specified using the `bazel_version` attribute, or the `bazel_versions` parameter to `bazel_integration_tests`? Or alternatively by setting `bazel_binary` explicitly?

Reply:
You are absolutely correct. I was thinking about `bazel_binary` when I answered your question, not `bazel_binaries`. (In retrospect, that was a very poor name on my part. 😳)

With regard to `bazel_binaries`, I will try loading it in the file instead of passing it as a parameter.

Reply:
After playing around with this, I remember why I passed the `bazel_binaries` into `bazel_integration_test`. We need the `@bazel_binaries//:defs.bzl` to be evaluated in the client's workspace, not `rules_bazel_integration_test`. 

I refactored the code to remove the `bazel_binaries` parameter and added the following to `bazel_integration_tests.bzl`:
```pytyon
load("@bazel_binaries//:defs.bzl", "bazel_binaries")
```

When a client repo attempts to load `rules_bazel_integration_test`, the following error occurs:
```
ERROR: Skipping '//...': error loading package under directory '': error loading package 'tests/updatesrc_tests': at /private/var/tmp/_bazel_chuck/a0f4fd87a92e40b507a5e30228e852a8/external/rules_bazel_integration_test~override/bazel_integration_test/defs.bzl:8:5: at /private/var/tmp/_bazel_chuck/a0f4fd87a92e40b507a5e30228e852a8/external/rules_bazel_integration_test~override/bazel_integration_test/private/bazel_integration_test.bzl:5:6: Unable to find package for @[unknown repo 'bazel_binaries' requested from @rules_bazel_integration_test~override]//:defs.bzl: The repository '@[unknown repo 'bazel_binaries' requested from @rules_bazel_integration_test~override]' could not be resolved: No repository visible as '@bazel_binaries' from repository '@rules_bazel_integration_test~override'.
```

The client does define a `@bazel_binaries` repo. This behavior seems correct in that we want the `@bazel_binaries` from the client workspace and not from the `@rules_bazel_integration_test` workspace. (I hope that I have that terminology correct.)

@aherrmann @fmeum I created [a draft PR](https://github.com/bazel-contrib/rules_bazel_integration_test/pull/139) with my changes. Perhaps I missed something in my implementation. Please let me know if you see any issues.